src/output-writer/google-sheets-writer/index.ts

```diff
@@ -0,0 +1,209 @@
+        sheetId: spreadsheet.data.sheets[0].properties.sheetId,
+      },
+    },
+    ...getInitialFormatting(sheetId, runInfoSheetId),
+  ]);
+
+  await writeRow(service, spreadsheetId, sheetNames.main, 1, [
+    '',
+    'Waiting for first Lighthouse report to complete...',
+  ]);
+
+  return {
+    async writeRunInfo(runInfo) {
+      const valuesBatchUpdateRequest: Sheets.sheets_v4.Schema$BatchUpdateValuesRequest =
+        {
+          data: [
+            {
+              range: `'${sheetNames.runInfo}'!A1`,
+              values: [
+                ['Command', runInfo.command],
+                ['Time', runInfo.time],
+                [],
+                ['node version:', runInfo.versions.node],
+                ['npm version:', runInfo.versions.npm],
+                ['lighthouse version:', runInfo.versions.lighthouse],
+                [
+                  'lighthouse-parade version:',
+                  runInfo.versions.lighthouseParade,
+                ],
+                ['chrome version:', runInfo.versions.chrome],
+                [],
+                ['OS:', runInfo.system.operatingSystem],
+                ['Memory:', runInfo.system.memory],
+                ['CPU Info', runInfo.system.cpus],
+              ],
+            },
+          ],
+        };
+
+      await service.spreadsheets.values
+        .batchUpdate({
+          spreadsheetId,
+          resource: valuesBatchUpdateRequest,
+          valueInputOption: 'USER_ENTERED',
+        } as any)
+        .then((r) => r);
+    },
+    async writeHeader(columns) {
+      mutexPromise = mutexPromise.then(async () => {
+        await writeRow(service, spreadsheetId, sheetNames.main, ++rowNum, [
+          '',
+          ...columns.map((c) => c.lighthouseCategory),
+        ]);
+        await writeRow(service, spreadsheetId, sheetNames.main, ++rowNum, [
+          'URL',
+          ...columns.map(
+            (c) => c.name + (c.nameDetail ? `\n(${c.nameDetail})` : '')
+          ),
+        ]);
+
+        const formattingBatchUpdates = getFormattingUpdates(sheetId, columns);
+        const {
+          batchUpdates: chartBatchUpdates,
+          batchValues: chartBatchValues,
+        } = getCharts(sheetId, auditBreakdownSheetId, columns);
+
+        await batchUpdate(service, spreadsheetId, [
+          ...formattingBatchUpdates,
+          ...chartBatchUpdates,
+        ]);
+
+        await service.spreadsheets.values
+          .batchUpdate({
+            spreadsheetId,
+            resource: chartBatchValues,
+            valueInputOption: 'USER_ENTERED',
+          } as any)
+          .then((r) => r);
+      });
+
+      await mutexPromise;
+    },
+    async addEntry(url, rowValues) {
+      mutexPromise = mutexPromise.then(async () => {
+        await writeRow(service, spreadsheetId, sheetNames.main, ++rowNum, [
+          url,
+          ...rowValues,
+        ]);
+      });
+
+      await mutexPromise;
+    },
+    async complete() {},
```

Comment:
Is this meant to be empty? 🤔

Reply:
Yeah, in this case it didn't need to do anything in the `complete` phase (unlike the CSV writer which has to close the file handle)

Reply:
Got it, thanks. Can you add that as a comment? 🙂